Validation log does not show \n properly

Validation logs are stored as a chain, using \n between lines. Current element does not show them but a space.

- [x] Show \n as new lines
- [ ] Use black background
- [ ] Show [OK] starting lines in green
- [ ] Show [ERROR] starting lines in red
- [ ] Show [WARNING] starting lines in orange

--- a/src/actions/index.js
+++ b/src/actions/index.js
@@ -6,10 +6,15 @@ export function signInSuccess(user) {
     user: user
   };
 }
-export function signInError(user) {
+export function signInError(error) {
   return {
     type: actions.SIGN_IN_ERROR,
-    user: user
+    error: error
+  };
+}
+export function notSignIn() {
+  return {
+    type: actions.NOT_SIGN_IN
   };
 }
 export function signInLoading(isLoading) {
@@ -27,7 +32,7 @@ export function getUsersSuccess(users) {
 export function getUsersError(error) {
   return {
     type: actions.GET_USERS_ERROR,
-    error:error
+    error: error
   };
 }
 export function getUserSuccess(user) {
@@ -54,9 +59,10 @@ export function getDataAPiSuccess(dataApi) {
   };
 }
 
-export function getDataAPiError() {
+export function getDataAPiError(error) {
   return {
-    type: actions.DATA_API_ERROR
+    type: actions.DATA_API_ERROR,
+    error: error
   };
 }
 

--- a/src/actions/typeActions.js
+++ b/src/actions/typeActions.js
@@ -1,5 +1,6 @@
 export const SIGN_IN_SUCCESS = 'SIGN_IN_SUCCESS';
 export const SIGN_IN_ERROR = 'SIGN_IN_ERROR';
+export const NOT_SIGN_IN = 'NOT_SIGN_IN';
 export const SIGN_IN_LOADING = 'SIGN_IN_LOADING';
 export const SIGN_OUT = 'SIGN_OUT';
 export const DATA_API_SUCCESS = 'DATA_API_SUCCESS';

--- a/src/components/Footer/index.js
+++ b/src/components/Footer/index.js
@@ -21,46 +21,95 @@ export default class Footer extends React.Component {
           </Col>
           <Col md={4} className="left-border">
             <section>
-              <h3>PRODUCTS</h3>
+              <h3 />
               <ul>
-                <li>Ansible Tower</li>
-                <li>Tower Editions</li>
-                <li>Tower pricing</li>
-                <li>Free Tower Trial</li>
-                <li>Ansible Consulting</li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="http://www.manageiq.org/"
+                    target="_blank"
+                  >
+                    ManageIQ website
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="http://talk.manageiq.org/"
+                    target="_blank"
+                  >
+                    ManageIQ Talk
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://github.com/ManageIQ"
+                    target="_blank"
+                  >
+                    ManageIQ in GitHub
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://github.com/ManageIQ-Exchange"
+                    target="_blank"
+                  >
+                    ManageIQ-Exchange in GitHub
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://www.redhat.com/en/technologies/management/cloudforms"
+                    target="_blank"
+                  >
+                    CloudForms
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://gitter.im/manageiq"
+                    target="_blank"
+                  >
+                    ManageIQ in Gitter
+                  </a>
+                </li>
               </ul>
             </section>
             <section>
-              <h3>RESOURCES</h3>
+              <h3 />
               <ul>
-                <li>Ansible Quick Start Guide</li>
-                <li>Videos</li>
-                <li>Webinars & Training</li>
-                <li>Case Studies</li>
-                <li>Whitepapers</li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://github.com/ManageIQ-Exchange/manageiq-exchange-frontend/issues"
+                    target="_blank"
+                  >
+                    Frontend issues
+                  </a>
+                </li>
+                <li>
+                  <a
+                    className="link-index"
+                    href="https://github.com/ManageIQ-Exchange/manageiq-exchange/issues"
+                    target="_blank"
+                  >
+                    API issues
+                  </a>
+                </li>
               </ul>
             </section>
           </Col>
           <Col md={4} className="left-border">
             <section>
-              <h3>COMMUNITY</h3>
-              <ul>
-                <li>Upcoming Events</li>
-                <li>Ansible on Meetup</li>
-                <li>Ansible on Github</li>
-              </ul>
-
-              <ul>
-                <li>ANSIBLE SUPPORT</li>
-                <li>ANSIBLE DOCS</li>
-                <li>ANSIBLE STORE</li>
-                <li>ANSIBLE BLOG</li>
-              </ul>
-
+              <h3 />
+              <ul />
+              <ul />
               <div className="content-icon">
-                <a
-                  href="https://github.com/miq-consumption/manageiq-EXCHANGE"
-                >
+                <a href="https://github.com/ManageIQ-Exchange/manageiq-exchange/">
                   <Icon
                     style={{ color: COLOUR_LOGO }}
                     name="twitter-square fa-3x"

--- a/src/components/Footer/style.css
+++ b/src/components/Footer/style.css
@@ -29,3 +29,6 @@ ul {
 .content-icon * {
   margin-right: 5px;
 }
+.link-index{
+  color: #FFFFFF;
+}

--- a/src/components/Layout.js
+++ b/src/components/Layout.js
@@ -1,18 +1,38 @@
 import React from 'react';
 import Menu from './Nav/Menu';
-import { AboutModal } from 'patternfly-react';
+import { AboutModal, Alert } from 'patternfly-react';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
 import Login from './Login/';
 import { signIn, checkSessionUser, signOut } from '../thunk/user';
 import { apiVersion } from '../thunk/dataApi';
+import { User } from '../models/user';
 
+const defaultProps = {
+  apiVersion: {
+    dataApi: {},
+    error: null
+  },
+  user: {
+    user: new User(),
+    error: null
+  }
+};
+const propTypes = {
+  checkSessionUser: PropTypes.func,
+  getApiVersion: PropTypes.func,
+  user: PropTypes.object,
+  apiVersion: PropTypes.object
+};
 class Layout extends React.Component {
   constructor(props) {
     super(props);
     this.isShowModal = this.isShowModal.bind(this);
     this.onClickLogin = this.onClickLogin.bind(this);
     this.state = {
-      showModalLogin: false
+      showModalLogin: false,
+      showAlertAlready: false
     };
   }
   componentDidMount() {
@@ -31,31 +51,59 @@ class Layout extends React.Component {
     this.props.signIn(code, provider);
     this.isShowModal(false);
   }
-
+  changeShowAlert = () => {
+    this.setState({ showAlertAlready: true });
+  };
+  renderAlert = () => {
+    const { user, apiVersion } = this.props;
+    const { showAlertAlready } = this.state;
+    const message = 'There have been a problem with the server';
+    return !showAlertAlready ?
+      user.error || apiVersion.error ? (
+      <Alert
+        style={{
+          width: '30%',
+          position: 'absolute',
+          zIndex: 9999,
+          right: '2%'
+        }}
+        type="warning"
+        onDismiss={this.changeShowAlert}
+      >
+        {message}
+      </Alert>
+      ) : null
+      : null;
+  };
   render() {
     let { showModalLogin } = this.state;
-    let { user, apiVersion } = this.props;
+    const { user, apiVersion } = this.props;
+    const dataApi = apiVersion.dataApi;
     return (
       <div className="app-container">
         <header>
-          <Menu isShowModal={this.isShowModal} user={user} signOut={this.props.signOut}  />
+          <Menu
+            isShowModal={this.isShowModal}
+            user={user}
+            signOut={this.props.signOut}
+          />
         </header>
+        {this.renderAlert()}
         <AboutModal
           show={showModalLogin}
           onHide={() => this.isShowModal(false)}
           productTitle="Sign In to Your Account"
           trademarkText={null}
         >
-          <Login
-            onSignIn={this.onClickLogin}
-            provider={apiVersion.providers}
-          />
+          <Login onSignIn={this.onClickLogin} provider={dataApi.providers} />
         </AboutModal>
         <div className="content-layout">{this.props.children}</div>
       </div>
     );
   }
 }
+Layout.propTypes = propTypes;
+Layout.defaultProps = defaultProps;
 
 const mapStateToProps = state => {
   return {

--- a/src/components/ListRanking/index.js
+++ b/src/components/ListRanking/index.js
@@ -58,7 +58,7 @@ class ListRanking extends React.Component {
                         <Row style={{ width: '100%' }}>
                           <Col md={9} xs={9} className="col-header">
                             <a
-                              className="first-header"
+                              className="first-header link-first-header"
                               onClick={() =>
                                 this.onClick(data[idObject], data[keys[0]])
                               }

--- a/src/components/ListRanking/index.js
+++ b/src/components/ListRanking/index.js
@@ -76,7 +76,9 @@ class ListRanking extends React.Component {
                             }}
                           >
                             {this.props.children ?
-                              <a href={data[keys[1]]}>{this.props.children}</a>
+                              <a href={data[keys[1]]} target="_blank">
+                                {this.props.children}
+                              </a>
                              :
                               <span>{data[keys[1]]}</span>
                             }

--- a/src/components/ListRanking/style.css
+++ b/src/components/ListRanking/style.css
@@ -30,3 +30,6 @@ a {
 a:hover {
   text-decoration: none !important;
 }
+.link-first-header {
+  cursor: pointer;
+}

--- a/src/containers/MyContentPage/ListViewItem/index.js
+++ b/src/containers/MyContentPage/ListViewItem/index.js
@@ -20,6 +20,16 @@ class ListViewItem extends React.Component {
     const { openDetails } = this.state;
     this.setState({ openDetails: !openDetails });
   };
+  renderLog(log) {
+    return log.split('\n').map((item, key) => {
+      return (
+        <span key={key}>
+          {item}
+          <br />
+        </span>
+      );
+    });
+  }
   render() {
     const { openDetails } = this.state;
     const { full_name, id, published, validation_log } = this.props.dataList;
@@ -83,7 +93,7 @@ class ListViewItem extends React.Component {
             <h3>
               <strong>Validation log:</strong>
             </h3>
-            <span>{validation_log}</span>
+            <span>{this.renderLog(validation_log)}</span>
           </div>
         </Collapse>
       </div>

--- a/src/containers/MyContentPage/index.js
+++ b/src/containers/MyContentPage/index.js
@@ -29,9 +29,7 @@ import * as constant from './constant';
 const defaultProps = {
   spins: {
     spins: []
-  },
-  showAlert: false,
-  messageAlert: ''
+  }
 };
 
 const propTypes = {
@@ -58,8 +56,8 @@ export class MyContentPage extends React.Component {
       loadingPublish: false,
       listSpins: props.spins.spins,
       listSpinsComplete: props.spins.spins,
-      showAlert: props.showAlert,
-      messageAlert: props.messageAlert
+      showAlert: false,
+      messageAlert: ''
     };
   }
   componentDidMount() {

--- a/src/containers/SearchPage/style.css
+++ b/src/containers/SearchPage/style.css
@@ -17,10 +17,3 @@
   overflow-y: auto;
   height:700px;
 }
-.alert {
-  box-shadow: 0 2px 2px rgba(0,0,0,0.19), 0 2px 2px rgba(0,0,0,0.23);
-  width: 100%;
-}
-.alert:hover {
-  box-shadow: 0 8px 8px rgba(0,0,0,0.19), 0 8px 8px rgba(0,0,0,0.23);
-}

--- a/src/containers/SpinProfile/TabDetails/index.js
+++ b/src/containers/SpinProfile/TabDetails/index.js
@@ -76,7 +76,7 @@ export default class TabDetails extends React.Component {
             <Col md={12} className="content-details">
               <div>
                 <span>
-                  <strong>Licencia</strong>
+                  <strong>Software License</strong>
                 </span>
                 <span style={{ float: "right" }}> {licence}</span>
               </div>

--- a/src/index.css
+++ b/src/index.css
@@ -52,3 +52,10 @@ a:hover {
   width: 100%;
   height: 100%;
 }
+.alert {
+  box-shadow:0 2px 2px 0 rgba(0, 0, 0, 0.16), 0 0 0 1px rgba(0, 0, 0, 0.08);
+  width: 100%;
+}
+.alert:hover {
+  box-shadow: 0 6px 6px rgba(0,0,0,0.19), 0 5px 5px rgba(0,0,0,0.23);
+}

--- a/src/reducers/apiVersionReducer.js
+++ b/src/reducers/apiVersionReducer.js
@@ -3,15 +3,21 @@ import {
   DATA_API_ERROR
 } from '../actions/typeActions';
 
-export const apiVersion = (apiVersion = {}, action: any) => {
-  let newState = {};
+const stateDefault = {
+  dataApi: {},
+  error: null
+};
+
+export const apiVersion = (apiVersion = stateDefault, action: any) => {
+  let newState = Object.assign({}, apiVersion);
   switch (action.type) {
     case DATA_API_SUCCESS:
-      newState = Object.assign({}, action.dataApi);
+      newState.dataApi = Object.assign({}, action.dataApi);
       return newState;
     case DATA_API_ERROR:
-      return apiVersion;
+      newState.error = action.error;
+      return newState;
     default:
-      return apiVersion;
+      return newState;
   }
 };

--- a/src/reducers/userReducer.js
+++ b/src/reducers/userReducer.js
@@ -2,29 +2,36 @@ import {
   SIGN_IN_ERROR,
   SIGN_IN_SUCCESS,
   SIGN_OUT,
-  SIGN_IN_LOADING
+  SIGN_IN_LOADING,
+  NOT_SIGN_IN
 } from '../actions/typeActions';
 import { User } from '../models/user';
 
-const initState={
+const initState = {
   user: new User(),
-  loading:false,
-  logged: false
-}
+  loading: false,
+  logged: false,
+  error: null
+};
 export const user = (user = initState, action: any) => {
   let newState = Object.assign({}, user);
   switch (action.type) {
     case SIGN_IN_SUCCESS:
       newState.user = Object.assign({}, action.user);
       newState.logged = true;
+      newState.error = null;
       return newState;
     case SIGN_IN_ERROR:
       newState.logged = false;
+      newState.error = action.error;
       return newState;
     case SIGN_OUT:
       return initState;
     case SIGN_IN_LOADING:
       newState.loading = action.isLoading;
+      return newState;
+    case NOT_SIGN_IN:
+      newState.logged = false;
       return newState;
     default:
       return user;

--- a/src/thunk/dataApi.js
+++ b/src/thunk/dataApi.js
@@ -8,7 +8,7 @@ export function apiVersion() {
         dispatch(getDataAPiSuccess(response.data.data))
       })
       .catch(error => {
-        dispatch(getDataAPiError());
+        dispatch(getDataAPiError(error));
       });
   };
 }

--- a/src/thunk/user.js
+++ b/src/thunk/user.js
@@ -1,4 +1,4 @@
-import { sessionUserDataSave } from "../components/SocialButtonLogin/utils";
+import { sessionUserDataSave } from '../components/SocialButtonLogin/utils';
 import {
   signInSuccess,
   signInError,
@@ -20,10 +20,11 @@ import {
   getUserSuccess,
   getUserError,
   validateSpinSuccess,
-  validateSpinError
-} from "../actions/index";
-import { recoverUser, deleteUser } from "../storage/";
-import Api from "../service/Api";
+  validateSpinError,
+  notSignIn
+} from '../actions/index';
+import { recoverUser, deleteUser } from '../storage/';
+import Api from '../service/Api';
 
 export function signIn(code, provider) {
   return dispatch => {
@@ -54,7 +55,7 @@ export function checkSessionUser() {
     let user = recoverUser();
     if (user.github_login && user.github_login !== '')
       dispatch(signInSuccess(user));
-    else dispatch(signInError());
+    else dispatch(notSignIn());
   };
 }
 

